radar-upload-backend/src/main/java/org/radarbase/upload/resource/RecordResource.kt

```diff
@@ -172,17 +173,25 @@ class RecordResource {
     @Path("{recordId}/contents/{fileName}")
     fun getContents(
             @PathParam("fileName") fileName: String,
-            @PathParam("recordId") recordId: Long): Response {
+            @PathParam("recordId") recordId: Long,
+            @Context config: Config): Response {
 
         val recordContent = recordRepository.readRecordContent(recordId, fileName)
                 ?: throw NotFoundException("Cannot find record content with record-id $recordId and fileName $fileName")
 
-        val content = recordRepository.readFileContent(recordId, fileName)
-                ?: throw NotFoundException("Cannot read content of file $fileName from record $recordId")
+        logger.info("Reading record $recordId file $fileName of size ${recordContent.size}")
 
-        val streamingOutput = StreamingOutput {
-            content.inputStream().use { inStream -> inStream.copyTo(it) }
-            it.flush()
+        val streamingOutput = StreamingOutput { out ->
+            for (position in 0L .. recordContent.size step config.contentStreamBufferSize) {
+                val len = config.contentStreamBufferSize.coerceAtMost(recordContent.size - position)
+                if (len > 0) {
+                    val content = recordRepository.readFileContent(recordId, recordContent.record.metadata.revision, recordContent.fileName, position, len)
```

Comment:
this would create new transaction for each read. I guess that was intentional?

Reply:
Yes, this goes around the streaming exceptions that we got when reading data outside of a transaction. The buffer size is chosen to be quite large to avoid too much performance degradation.

Reply:
Changed to a single transaction model.